src/main/java/org/folio/innreach/batch/contribution/service/InitialContributionEventProcessor.java

```diff
@@ -0,0 +1,178 @@
+      try {
+        var instanceId = job.getInstanceId();
+        var centralServerId = contributionRecord.get(job.getJobId()) != null ?
+          contributionRecord.get(job.getJobId()).getCentralServer().getId() : getCentralServerId(job.getJobId());
+        var instance = inventoryViewService.getInstance(instanceId);
+        if (centralServerId == null || instance == null) {
+          log.warn("processInitialContributionEvents:: Unable to process event with instance " +
+            "id {} centralServerId {} ", instanceId, centralServerId);
+          updateJobAndContributionStatus(job, FAILED, job.isInstanceContributed());
+          return;
+        }
+        checkRetryLimit(job);
+        startContribution(centralServerId, instance, job);
+      } catch (ServiceSuspendedException | InnReachConnectionException | FeignException |
+               SocketTimeOutExceptionWrapper ex) {
+        log.warn("processInitialContributionEvents:: Retrying the contribution for instanceId {} due to {} ",
+          job.getInstanceId(), ex.getMessage());
+        updateJobAndContributionStatus(job, RETRY, job.isInstanceContributed());
+      } catch (Exception ex) {
+        log.warn("processInitialContributionEvents:: Exception while processing instanceId {} ", job.getInstanceId());
+        logException(job, ex, contributionRecord.get(job.getJobId()).getId());
+        updateJobAndContributionStatus(job, FAILED, job.isInstanceContributed());
+      }
+    });
+  }
+
+  private void checkRetryLimit(JobExecutionStatus job) {
+    if (maxRetryAttempts != 0 && job.getRetryAttempts() > maxRetryAttempts) {
+      log.warn("checkRetryLimit:: Retry limit exhausted for instanceId {} ", job.getInstanceId());
+      throw new RetryException("Retry limit exhausted");
+    }
+  }
+
+  private UUID getCentralServerId(UUID jobId) {
+    Contribution contribution = contributionRepository.findByJobId(jobId);
+    if (contribution != null && contribution.getCentralServer() != null) {
+      contributionRecord.put(jobId, contribution);
+      return contribution.getCentralServer().getId();
+    }
+    return null;
+  }
+
+  private void startContribution(UUID centralServerId, Instance instance, JobExecutionStatus job) throws SocketTimeoutException {
+    var instanceId = job.getInstanceId();
+    if (isEligibleForContribution(centralServerId, instance)) {
+      if(job.isInstanceContributed()){
+        log.info("startContribution:: Item contribution started for centralServerId: {}, instanceId: {}", centralServerId, instanceId);
+        contributeItem(job, centralServerId, instance);
+      } else {
+        log.info("startContribution:: Instance contribution started for instanceId {} ", instance.getId());
+        recordContributionService.contributeInstanceWithoutRetry(centralServerId, instance);
+        updateJobAndContributionStatus(job, READY, true);
+      }
+    } else if (isContributed(centralServerId, instance)) {
```

Comment:
Let's discuss. Wouldn't this have the effect of de-contributing anything that had been contributed previously. We're running this to "shore up" an existing initial contribution job. Not sure if this is what we want. I will talk to Brooks about this. But let's make sure to discuss before we roll this out.

Reply:
Ok I verified that this is in fact the correct logic with Brooks. The contribution eligibility can have changed since it was last contributed, and if it is no longer eligible, and has been contributed, it should be de-contributed.